fix(sn_transfers): be sure we store CashNotes before writing the wall…

joshuef opened the pull request:
…et file

This should prevent concurrency issues with referenced Notes not existing on disk yet

## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 05 Oct 23 10:18 UTC
This pull request fixes an issue related to concurrency when storing CashNotes in the wallet file. It ensures that CashNotes are stored before writing the wallet file to prevent any concurrency issues with referenced Notes not existing on disk yet. There are changes made in multiple files to address this issue.
<!-- reviewpad:summarize:end -->

maqi commented on sn/src/routing/core/msg_handling/join.rs:
```diff
@@ -94,20 +94,6 @@ impl Core {
             ]);
         }
 
-        if self.dkg_voter.has_dkg_underway() {
```
not sure about removing this toally.
As I understand, we shall change to: accept JoinRequest with aggregated OnlineApproval even has ongoing dkg.

joshuef replied:
I don't think we need it for anything else? To my mind it serves to prevent issues at signature aggregation which we solve otherwise in this PR.

bochaco commented on sn/src/messaging/system/join.rs:
```diff
@@ -103,5 +103,5 @@ pub enum JoinRejectionReason {
     NodeNotReachable(SocketAddr),
     /// Backoff as the section is undergoing DKG
-    DKGUnderway,
+    DkgUnderway,
```
Shouldn't this be removed?

joshuef committed:
fix(sn_transfers): be sure we store CashNotes before writing the wallet file

This should prevent concurrency issues with referenced Notes not existing on disk yet

grumbach reviewed (approved):
LGTM!
